group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1440,9 +1508,20 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
                 records = consumerGroupFenceMember(group, member);
             }
         }
-        return new CoordinatorResult<>(records, new ConsumerGroupHeartbeatResponseData()
-            .setMemberId(memberId)
-            .setMemberEpoch(memberEpoch));
+
+        CompletableFuture<Void> appendFuture = null;
+        if ((instanceId == null || memberEpoch != LEAVE_GROUP_STATIC_MEMBER_EPOCH) &&
```

Comment:
Instead of repeating the conditions, could we use a boolean that we set in the relevant places?